weaver/cli.py

```diff
@@ -289,7 +307,10 @@ def response_parser(self, response):
 
     def __call__(self, request):
         # type: (AnyRequestType) -> AnyRequestType
-        auth_token = self.request_auth()
+        if self.token is None:
+            auth_token = self.token
+        else:
+            auth_token = self.request_auth()
```

Comment:
Isn't it supposed to be the other way around, such that the request is performed if the token was not pre-filled?

Reply:
If the token was retrieved from the request, we should probably call `parse_token` below, just before calling `auth_header`. Otherwise, we get a situation where the handler could have multiple representations of the token depending on whether it came predefined or from the request.

Reply:
> Isn't it supposed to be the other way around,

Yes... that's embarrassing. It was the other way when I wrote the tests otherwise they would have all failed. I don't know what happened. I'll fix it